### PR TITLE
Rounding large numbers

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes BX-1550, BX-121, BX-1551

## What changed:
- this PR effects the swap review cards and the swap review
- follow up PRs to follow changing this behavior elsewhere in the app
- numbers >= 1,000 are now abbreviated using K, M, B, T, etc.

## Screen recordings / screenshots
New behavior:
Screenshot 2024-07-02 at 11 07 26 PM
Screenshot 2024-07-02 at 11 05 58 PM
Screenshot 2024-07-02 at 11 08 23 PM
Screenshot 2024-07-09 at 3 46 58 PM

## What to test
- do large and small values round correctly on the review cards

### CODE REVIEW

**Review by estebanmino — COMMENTED** [review]
i think @greg-schrammel already did this before ?

image

**Comment by greg-schrammel** [comment]
yep formatNumber takes an options arg that you can add `notation: 'compact'`

```ts
import { formatCurrency } from '~/core/utils/formatNumber';

formatCurrency(num, { notation: 'compact', maximumSignificantDigits: 4 });
```

File: src/entries/popup/pages/swap/SwapReviewSheet/SwapReviewSheet.tsx [hunk 1]
```diff
@@ -243,6 +244,33 @@ const SwapReviewSheetWithQuote = ({
...
+  const processArray = (arr: string[]): string[] => {
+    return arr.map((item) => {
+      const parts = item.split(' ');
+      if (parts.length === 5) {
+        const formattedAmount = formatNumber(parts[3]);
+        return `${parts[0]} ${parts[1]} ${parts[2]} ${formattedAmount} ${parts[4]}`;
+      }
+      return item;
+    });
+  };
```
Comment (derHowie) [inline src/entries/popup/pages/swap/SwapReviewSheet/SwapReviewSheet.tsx:266]: I think this type of functionality should be dropped into a util file. Wdyt?

File: src/entries/popup/pages/swap/SwapReviewSheet/SwapAssetCard.tsx [hunk 2]
```diff
@@ -49,6 +50,18 @@ export const SwapAssetCard = ({
...
+      return parseFloat(n.replace(/,/g, ''));
+    }
+    return n || 0;
+  };
+
+  const formatNumber = (n?: number | string | null) =>
+    formatCurrency(cleanNumber(n), {
+      notation: 'compact',
+      maximumSignificantDigits: 4,
+    });
```
Comment (derHowie) [inline src/entries/popup/pages/swap/SwapReviewSheet/SwapAssetCard.tsx:64]: same here

File: src/entries/popup/pages/swap/SwapReviewSheet/SwapReviewSheet.tsx (hunk 1 above)
Comment (BrodyHughes) [reply src/entries/popup/pages/swap/SwapReviewSheet/SwapReviewSheet.tsx:266]: yeah that sounds good to me!

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
It looks like we're accidentally appending a `$` to the swap bubble values
Screenshot 2024-07-11 at 6 35 17 PM

**Comment by BrodyHughes** [comment]
> It looks like we're accidentally appending a `$` to the swap bubble values Screenshot 2024-07-11 at 6 35 17 PM

fixed this concern. nice catch.

**Review by derHowie — APPROVED** [review]
👌

**Review by DanielSinclair — APPROVED** [review]

**Pull request merged**